pagination/pagination.go

```diff
@@ -0,0 +1,129 @@
+package pagination
```

Comment:
No doc comments for this package (to appear in `godoc`)

Would be good to have something similar to the README in here (annoying it is duplicated, perhaps everything should be `godoc`s). Should have plenty of user based examples too.